controllers/serviceexport_controller.go

```diff
@@ -129,18 +130,17 @@ func (r *ServiceExportReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	if srvExport.ObjectMeta.Annotations["multicluster.x-k8s.io/federation"] == "amazon-vpc-lattice" {
-		r.log.Infof("ServiceExportReconciler --- found matching service export --- %s\n", srvExport.Name)
+		r.log.Debugf("Found matching service export %s-%s", srvExport.Name, srvExport.Namespace)
 
 		if !srvExport.DeletionTimestamp.IsZero() {
-			r.log.Info("Deleting")
 			if err := r.cleanupServiceExportResources(ctx, srvExport); err != nil {
-				r.log.Infof("Failed to clean up service export %v, err :%v \n", srvExport, err)
 				return err
 			}
-
-			r.log.Info("Successfully delete")
-
-			r.finalizerManager.RemoveFinalizers(ctx, srvExport, serviceExportFinalizer)
+			err := r.finalizerManager.RemoveFinalizers(ctx, srvExport, serviceExportFinalizer)
+			if err != nil {
+				r.log.Errorf("Failed to remove finalizers for service export %s-%s due to %s",
+					srvExport.Name, srvExport.Namespace, err)
```

Comment:
I think you forgot to return `err` here

Reply:
This didn't previously return the err, and we don't do it in most of our other controllers. I'd rather avoid a behaviour change in a logging-focused PR

Reply:
oh I mis-read the diff